### PR TITLE
VOS by Dejian Zou

### PR DESCRIPTION
**Pull request by JediWarriorZou**
I have finished implementing Virtual Outlier Systhesis. This method generates virtual outlier to help train the network.
The network is WideResnet , train_dataset is cifar10 and OOD dataset for testing is svhn.The metrics are shown in the table.
| Metric        | Value/% |
|---------------|---------|
| VAL_ACC       | 94.55   |
| fpr95         | 47.71   |
| AUROC         | 93.73   |
| AUPR_IN       | 91.45   |
| AUPR_OUT      | 96.40   |
| DETECTION_ERR | 11.56   |

### CODE REVIEW

File: configs/networks/conf_esti.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 network:
   name: conf_wideresnet
   num_classes: '@{dataset.num_classes}'
-  pretrained: True # set 'True' to load pretrained model
+  pretrained: False # set 'True' to load pretrained model
```
Comment (Jingkang50) [inline configs/networks/conf_esti.yml:4]: Do we need pretrained model for conf?

File: openood/networks/resnet18_32x32.py [hunk 2]
```diff
@@ -107,7 +117,7 @@ def _make_layer(self, block, planes, num_blocks, stride):
             self.in_planes = planes * block.expansion
         return nn.Sequential(*layers)
 
-    def forward(self, x, return_feature=False, return_feature_list=False):
+    def forward(self, x, return_feature_list=False):
```
Comment (Jingkang50) [inline openood/networks/resnet18_32x32.py:120]: why delete `return_feature` option since other methods might require

File: openood/preprocessors/draem_preprocessor.py [hunk 3]
```diff
@@ -3,7 +3,7 @@
 import os
 
 import cv2
-import imgaug.augmenters as iaa
+#import imgaug.augmenters as iaa
```
Comment (Jingkang50) [inline openood/preprocessors/draem_preprocessor.py:6]: why comment it out?

**Review by Jingkang50 — CHANGES_REQUESTED** [review]
please revise accordingly

**Comment by Jingkang50** [comment]
Please make sure not to delete / comment out others' code before your future pull request.

File: openood/networks/resnet18_32x32.py (hunk 2 above)
Comment (JediWarriorZou) [reply openood/networks/resnet18_32x32.py:120]: I mimic wrn to put feature in feature_list. If it is not needed, it can remain the original form since VOS doesn't use resnet 32

File: configs/networks/conf_esti.yml (hunk 1 above)
Comment (JediWarriorZou) [reply configs/networks/conf_esti.yml:4]: In training pipeline, pretrain is False. In testing pipeline, pretrain is true

File: openood/preprocessors/draem_preprocessor.py (hunk 3 above)
Comment (JediWarriorZou) [reply openood/preprocessors/draem_preprocessor.py:6]: Oh, I forget to remove the"#"

File: openood/networks/resnet18_32x32.py [hunk 4]
```diff
@@ -122,7 +132,7 @@ def forward(self, x, return_feature=False, return_feature_list=False):
         elif return_feature_list:
             return logits_cls, feature_list
         else:
-            return logits_cls
```
Comment (Jingkang50) [inline openood/networks/resnet18_32x32.py:125]: why delete

**Review by Jingkang50 — CHANGES_REQUESTED** [review]

**Review by Jingkang50 — APPROVED** [review]
